src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -2035,6 +2036,19 @@ because xbuild doesn't support framework reference assemblies.
     <Touch Files="$(_AndroidLinkFlag)" AlwaysCreate="true" />
 </Target>
 	
+<Target Name="_GenerateJniMarshalMethods"
+  Condition="'$(JniMarshalMethods)' == 'True' And '$(Configuration)' == 'Release' And '$(AndroidLinkMode)' != 'None'"
+  DependsOnTargets="_GetReferenceAssemblyPaths"
+  Inputs="@(ResolvedUserAssemblies->'$(MonoAndroidLinkerInputDir)%(Filename)%(Extension)')"
+  Outputs="$(_AndroidJniMarshalMethodsFlag)">
+
+    <Exec
+       Command="MONO_PATH=$(_XATargetFrameworkDirectories) $(MonoAndroidBinDirectory)\mono $(MonoAndroidBinDirectory)\..\jnimarshalmethod-gen.exe @(ResolvedUserAssemblies->'$(MonoAndroidLinkerInputDir)%(Filename)%(Extension)')"
```

Comment:
Don't we also need to update the [`GetMonoBundleItems` target within `src/mono-runtimes/mono-runtimes.targets`](https://github.com/xamarin/xamarin-android/blob/32adcc6926435455305aa65af8d271504f6e2cc5/src/mono-runtimes/mono-runtimes.targets#L795-L824) so that `mono` is included in the `bundle*.zip` file?

Reply:
`$(_XATargetFrameworkDirectories)` should be quoted, in case it contains spaces:

```xml
Command="MONO_PATH=&quot;$(_XATargetFrameworkDirectories)&quot; ...
```

Additionally, I believe that `$(_XATargetFrameworkDirectories)` will be *incomplete*: it should contain only the directories (directory) for `MonoAndroid,Version=v1.0`, which will *not* be a directory which contains `Mono.Android.dll`.

For example (from a random full xamarin-android build log I have lying around):

```
Task "ResolveSdks" (TaskId:1387)
...
  Task Parameter:ReferenceAssemblyPaths=/Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/bin/Debug/lib/xamarin.android/xbuild-frameworks/MonoAndroid/v1.0/ (TaskId:1387)
...
  Output Property: _XATargetFrameworkDirectories=/Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/bin/Debug/lib/xamarin.android/xbuild-frameworks/MonoAndroid/v1.0/ (TaskId:1387)
```

If e.g. `.../xbuild-frameworks/MonoAndroid/v8.1` isn't included, how will `Mono.Android.dll` be found?

*Alternatively*, perhaps `MONO_PATH` should be set to `$(MonoAndroidLinkerInputDir)` and called after that directory has been setup, e.g. after `_CopyIntermediateAssemblies`?